draft-ietf-quic-transport.md

```diff
@@ -454,30 +456,22 @@ Header Form:
 
 : The most significant bit (0x80) of octet 0 is set to 0 for the short header.
 
-Omit Connection ID Flag:
-
-: The second bit (0x40) of octet 0 indicates whether the Connection ID field is
-  omitted.  If set to 0, then the Connection ID field is present; if set to 1,
-  the Connection ID field is omitted.  The Connection ID field can
-  only be omitted if the omit_connection_id transport parameter
-  ({{transport-parameter-definitions}}) is specified by the intended recipient
-  of the packet.
-
 Key Phase Bit:
 
-: The third bit (0x20) of octet 0 indicates the key phase, which allows a
+: The second bit (0x40) of octet 0 indicates the key phase, which allows a
   recipient of a packet to identify the packet protection keys that are used to
   protect the packet.  See {{QUIC-TLS}} for details.
 
 Short Packet Type:
 
-: The remaining 5 bits of octet 0 include one of 32 packet types.
+: The remaining 6 bits of octet 0 include one of 64 packet types.
   {{short-packet-types}} lists the types that are defined for short packets.
 
 Connection ID:
 
-: If the Omit Connection ID Flag is not set, a connection ID occupies octets
-  1 through 8 of the packet.  See {{connection-id}} for more details.
+: A connection ID occupies between 0 and 17 octets inclusive, starting from
+  octet 1.  This value can be truncated from the full 17 octet connection ID.
+  See {{connection-id}} for more details.
 
```

Comment:
So, if I read correctly, there is no indication in the short header of the presence and size of the connection ID field. In practice, this forces clients and servers to pick one same size for all connections. Yes, in theory it can be negotiated on a per connection basis, but you need to parse the CID to retrieve the connection context and find the result of that negotiation. Better not try to change the encoded length over the lifetime of the server...

Reply:
A server has to pick a consistent minimum.  It could occasionally choose longer, but the load balancer would just ignore those extra bytes.